packages/core/src/machines/authMachine.ts

```diff
@@ -0,0 +1,172 @@
+import { assign, createMachine, forwardTo, spawn } from 'xstate';
+import { Auth, Amplify } from 'aws-amplify';
+import { AuthContext, AuthEvent } from '../types';
+import { inspect } from '@xstate/inspect';
+import {
+  signInActor,
+  signUpActor,
+  signOutActor,
+  resetPasswordActor,
+} from './actors';
+import { stopActor } from './actions';
+
+// TODO: Remove this before it's merged.
+if (typeof window !== 'undefined') {
+  inspect({
+    // options
+    // url: 'https://statecharts.io/inspect', // (default)
+    iframe: false, // open in new window
+  });
+}
+
+export const authMachine = createMachine<AuthContext, AuthEvent>(
+  {
+    id: 'auth',
+    initial: 'idle',
+    context: {},
+    states: {
+      // See: https://xstate.js.org/docs/guides/communication.html#invoking-promises
+      idle: {
+        invoke: [
+          {
+            // TODO Wait for Auth to be configured
+            src: 'getCurrentUser',
+            onDone: {
+              actions: 'setUser',
+              target: 'authenticated',
+            },
+            onError: 'signIn',
+          },
+          {
+            src: 'getAmplifyConfig',
+            onDone: {
+              actions: 'setAuthConfig',
+            },
+          },
+        ],
+      },
+      signIn: {
+        entry: 'spawnSignInActor',
+        exit: stopActor('signInActor'),
+        on: {
+          SIGN_UP: 'signUp',
+          RESET_PASSWORD: 'resetPassword',
+          'done.invoke.signInActor': [
+            {
+              target: 'signUp',
+              cond: 'shouldRedirectToSignUp',
+            },
+            {
+              target: 'authenticated',
+              actions: 'setUser',
+            },
+          ],
+        },
+      },
+      signUp: {
+        entry: 'spawnSignUpActor',
+        exit: stopActor('signUpActor'),
+        on: {
+          SIGN_IN: 'signIn',
+          'done.invoke.signUpActor': {
+            target: 'signIn',
+            actions: 'setUser',
+          },
+        },
+      },
+      resetPassword: {
+        entry: 'spawnResetPasswordActor',
+        exit: stopActor('resetPasswordActor'),
+        on: {
+          SIGN_IN: 'signIn',
+          'done.invoke.resetPasswordActor': 'signIn',
+        },
+      },
+      signOut: {
+        entry: 'spawnSignOutActor',
+        exit: [stopActor('signOutActor'), 'clearUser'],
+        on: { 'done.invoke.signOutActor': 'idle' },
+      },
+      authenticated: {
+        on: { SIGN_OUT: 'signOut' },
+      },
+    },
+    on: {
+      CHANGE: { actions: 'forwardToActor' },
+      SUBMIT: { actions: 'forwardToActor' },
+      FEDERATED_SIGN_IN: { actions: 'forwardToActor' },
+      RESEND: { actions: 'forwardToActor' },
+      SIGN_OUT: { actions: 'forwardToActor' },
```

Comment:
These are the events that are forwarded to the actors. We can do wildcard descriptor `*` here, but wanted to be explicit here to prevent communications from being too chatty.

Reply:
That's a fair point. Good thing is that there's no handler for `FEDERATED_SIGN_IN` in `resetPassword` flow, nor are there buttons in `resetPassword` screens able to trigger that event. But I see your point -- maybe this is worth moving those `forwardToActor` to each individual state. Adding to my TODOs.